system/util.rs

```diff
@@ -0,0 +1,212 @@
+pub mod transaction_util {
+    use concept::thing::thing_manager::ThingManager;
+    use concept::type_::type_manager::TypeManager;
+    use database::transaction::{DataCommitError, SchemaCommitError, TransactionRead, TransactionSchema, TransactionWrite};
+    use database::Database;
+    use function::function_manager::FunctionManager;
+    use lending_iterator::LendingIterator;
+    use options::TransactionOptions;
+    use query::query_manager::QueryManager;
+    use std::sync::Arc;
+    use storage::durability_client::WALClient;
+    use storage::snapshot::{SchemaSnapshot, WriteSnapshot};
+
+    #[derive(Debug)]
+    pub struct TransactionUtil {
+        database: Arc<Database<WALClient>>,
+    }
+
+    impl TransactionUtil {
+        pub fn new(database: Arc<Database<WALClient>>) -> Self {
+            TransactionUtil { database }
+        }
+
+        pub fn schema_transaction<T>(
+            &self,
+            fn_: impl Fn(
+                QueryManager,
+                &mut SchemaSnapshot<WALClient>,
+                &TypeManager,
+                &ThingManager,
+                &FunctionManager,
+            ) -> T
+        ) -> Result<T, SchemaCommitError> {
+            let TransactionSchema {
+                snapshot,
+                type_manager,
+                thing_manager,
+                function_manager,
+                database,
+                transaction_options,
+            } = TransactionSchema::open(self.database.clone(), TransactionOptions::default());
+            let mut snapshot: SchemaSnapshot<WALClient> = Arc::into_inner(snapshot).unwrap();
+            let result = fn_(QueryManager::new(), &mut snapshot, &type_manager, &thing_manager, &function_manager);
+            let tx = TransactionSchema::from(snapshot, type_manager, thing_manager, function_manager, database, transaction_options);
+            tx.commit().map(|u| result)
```

Comment:
```suggestion
            tx.commit().map(|_| result)
```

Reply:
Fixed in 08f3441d77d54a2fc5d18b90a24b199c00f5aeac.